kola/tests/docker/docker.go

```diff
@@ -459,17 +470,86 @@ func dockerUserNoCaps(c cluster.TestCluster) {
 	}
 }
 
-// testDockerInfo test that docker info's output is as expected.  the expected
-// filesystem may be asserted as one of 'overlay', 'btrfs', 'devicemapper'
-// depending on how the machine was launched.
-func testDockerInfo(expectedFs string, c cluster.TestCluster) {
+// dockerContainerdRestart ensures containerd will restart if it dies. It tests that containerd is running,
+// kills it, the tests that it came back up.
+func dockerContainerdRestart(c cluster.TestCluster) {
 	m := c.Machines()[0]
 
+	// verify containerd came up
+	if !testContainerdUp(c) {
+		c.Fatalf("Containerd did not start")
+	}
+
+	pid, err := m.SSH("systemctl show containerd -p MainPID")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if string(pid) == "MainPID=0" {
+		c.Fatalf("Could not find containerd pid")
+	}
+
+	// kill it
+	if _, err = m.SSH("sudo kill " + string(pid[8:])); err != nil {
+		c.Fatal(err)
+	}
+
+	// give it a chance to restart
+	if _, err = m.SSH("sleep 2"); err != nil {
+		c.Fatal(err)
+	}
+
+	// verify it came back and docker knows about it
+	if !testContainerdUp(c) {
+		c.Fatalf("Containerd did not restart")
+	}
+
+	// verify systemd started it and that it's pid is different
+	newPid, err := m.SSH("systemctl show containerd -p MainPID")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if string(newPid) == string(pid) {
+		c.Fatalf("Old and new pid's are the same. containerd did not die")
+	}
+}
+
+func testContainerdUp(c cluster.TestCluster) bool {
+	type containerdInfo struct {
```

Comment:
Let's move the struct to file scope, rather than redefining parts of it inside different tests.